pkg/controllers/osp/osp_controller.go

```diff
@@ -52,10 +53,11 @@ type Reconciler struct {
 	log             *zap.SugaredLogger
 	defaultOSPFiles map[string][]byte
 
-	namespace string
+	namespace    string
+	ospNamespace string
```

Comment:
Why this is needed? within the controller, you only need one namespace which the cluster namespace in case of Seed. In case of K1 there is only one namespace to use. Also the ospNamespace isn't use in the whole package. Please remove it.

Reply:
This is being used in the controller PTAL. 

In the context of OSP controller, `namespace` represents the namespace in which the controller is running; this it would expect `machine-controller` to be up and running in that namespace since it will watch it's deployment.
`ospNamespace` represents the namespace in which OSPs are supposed to be created, which in case of KKP will be `kubermatic` and for K1 it will be `kube-system`, I presume.

Reply:
But that's the thing, again, OSP controller doesn't care about any reconciled resources in the worker cluster and it only care about seed. So here is the case, machine controller is running in the cluster namespace within the seed. OSP controller is running there as well and it will retrieve/create resources only there, only in that namespace. Thus one namespace is enough.

Reply:
Okay, that makes sense. I'll update it.